### ISSUE TITLE
[MySQL] ON CASCADE child operations are not replicated

### ISSUE THREAD

**Issue by AdamKatzDev** [Altinity/clickhouse-sink-connector#421]
Not sure if this is a well known caveat but:
https://bugs.mysql.com/bug.php?id=32506  -   Foreign key cascades do not appear when binlog_format = 'ROW'

I hoped to use this connector for a project with a lot of `ON CASCADE` in schema.
Probably the fact that `ON CASCADE` child operations are not replicated for MySQL should be mentioned somewhere in the connector documentation. There could even be some protection added to the connector code itself.

</br>

Other sources:
https://dev.mysql.com/doc/refman/8.0/en/innodb-and-mysql-replication.html
>Replication and CASCADE.  Cascading actions for InnoDB tables on the source are executed on the replica only if the tables sharing the foreign key relation use InnoDB on both the source and replica. This is true whether you are using statement-based or row-based replication.

https://code.openark.org/blog/mysql/things-that-dont-work-well-with-mysqls-foreign-key-implementation
>No binary log entries for cascaded writes

https://debezium.io/documentation/faq/
>Why don’t I see DELETE events in some cases?
>This may be caused by the usage of CASCADE DELETE statements. In this case the deletion events generated by the database [are not part of the binlog](https://dev.mysql.com/doc/refman/5.7/en/innodb-and-mysql-replication.html) and thus cannot be captured by Debezium.

**Comment by aadant** [comment]
yes this is a known edge case for debezium and MySQL.  here is a workaround.

https://stackoverflow.com/questions/57443976/debezium-does-not-capture-cascade-delete

**Comment by AdamKatzDev** [comment]
@aadant is there a complete solution for this issue? A script for Clickhouse or MySQL that keep the triggers in a synced state?
For some removing cascade operations is not an option. I was really surprised when stumbled upon this bug, it wasn't a known issue for me at least.

**Comment by aadant** [comment]
@AdamKatzDev it would be a custom solution on your end.

**Comment by aadant** [comment]
confirmed here 
https://code.openark.org/blog/mysql/things-that-dont-work-well-with-mysqls-foreign-key-implementation#more-8226

**Comment by aadant** [comment]
@AdamKatzDev I recently noticed this issue too : I think it is possible to implement a solution with Materialized Views, the equivalent of MySQL triggers, one MV per constraint, I did not test it thoroughly but that's the theory.

Example: Simulated ON DELETE CASCADE
1. Parent Table

```
CREATE TABLE parent (
    id UInt64,
    name String,
    deleted UInt8 DEFAULT 0
) ENGINE = ReplacingMergeTree(deleted)
ORDER BY id;
```
2. Child Table

```
CREATE TABLE child (
    id UInt64,
    parent_id UInt64,
    value String,
    deleted UInt8 DEFAULT 0
) ENGINE = ReplacingMergeTree(deleted)
ORDER BY (parent_id, id);
```
3. Materialized View to Cascade Deletes
We’ll create a materialized view that listens for updates in the parent and marks child rows as “deleted”.

```
CREATE MATERIALIZED VIEW cascade_delete_child
TO child
AS
SELECT
    id AS id,   -- keep child id
    parent_id,
    value,
    1 AS deleted
FROM child
INNER JOIN parent
    ON child.parent_id = parent.id
WHERE parent.deleted = 1;
```

**Comment by aadant** [comment]
@AdamKatzDev MySQL recently fixed this bug
https://blogs.oracle.com/mysql/no-more-hidden-changes-how-mysql-9-6-transforms-foreign-key-management
That’s an excellent news
